### DISCUSSION TITLE
Multiple AddHttpClient with different API endpoints

### DISCUSSION THREAD

**Discussion by memis1970** [PrismLibrary/Prism#2613] [Xamarin.Forms] [answered]
Hi,

I have a Xamarin Forms app using Prism with the following packages:

Prism.Forms.Extended

Prism.Unity.Extensions

I'm trying to register 2 different APIs using HttpClientFactory as follows:

```
protected override void RegisterTypes(IContainerRegistry containerRegistry)
{
    //Omitted Code

    containerRegistry.RegisterServices(serviceCollection =>
    {
        serviceCollection.AddHttpClient<IApiServiceOne, ApiServiceOne>(client =>
        {
            client.BaseAddress = new Uri("Base Address One Here");
        });
        serviceCollection.AddHttpClient<IApiServiceTwo, ApiServiceTwo>(client =>
        {
            client.BaseAddress = new Uri("Base Address Two Here");
        });
    });
}
```

...but onyl one Base Address is being set. It sets the base address fine in there is only one AddHttpClient call.

Any ideas why this would be and how to remedy?

Thanks in advance

**Comment by dansiegel** [comment, marked as answer]
Unity does not support the concept of:

```cs
container.Register<IService, ImplementationA>();
container.Register<IService, ImplementationB>();

container.Resolve<IEnumerable<IService>>();
```

As I recall this would be required for the HttpClientFactory. If you use DryIoc instead it should work.

**Reply by memis1970** [reply]
@dansiegel 

Hi Dan,

Thanks for your prompt reply. Will give it a try with Dryloc then and report back.

Much appreciated.

**Reply by memis1970** [reply]
Happy Dayz. Works 1st time!

Thanks a bunch!
